### DISCUSSION TITLE
renovatebot only targetting `@types/*` and not respecting `prConcurrentLimit`?

### DISCUSSION THREAD

**Discussion by leo-mendoza-qatalog** [renovatebot/renovate#16104] [Request Help]
Hi all,

I'm new to renovatebot, so am fairly convinced I have it misconfigured somehow due to the behavior in the title. 

We've had it running for the last 3 weeks and, after an initial bad config where we turned off 'update all non-major dependencies' by closing the PR created for branch `renovate/all-minor-patch` , it's since only really picked up `@types` upgrades. Additionally, we've bumped the `prConcurrentLimit` to 10 but only see 4 open at a time.

Would love some advice on where we went wrong with this configuration. Oh, and we're using the GitHub Hosted App.

//edit: forgot to add, we've been running dependabot in parallel and dependabot is picking up other package updates, so we know they're out in the ether

Thanks very much!

renovate.json:
```
{
  "extends": [
    "config:base",
    ":dependencyDashboard",
    "group:nodeJs",
    "schedule:daily"
  ],
  "baseBranches": ["develop"],
  "major": {
    "dependencyDashboardApproval": true,
    "labels": ["dependencies", "major"],
    "reviewers": ["team:frontend"]
  },
  "minor": {
    "labels": ["dependencies", "minor"]
  },
  "prConcurrentLimit": 10,
  "rangeStrategy": "bump",
  "rebaseWhen": "conflicted",
  "updatePinnedDependencies": false,
  "packageRules": [
    {
      "groupName": "frontend packages",
      "reviewers": ["team:frontend"],
      "packageNames": [
        "common",
        "eslint-plugin-qatalog",
        "language-dot-work",
        "marketing-site",
        "mosaiq-ui",
        "segment-functions",
        "toblerone-ui",
        "web-client"
      ]
    },
    {
      "groupName": "backend packages",
      "reviewers": ["team:backend"],
      "packageNames": [
        "api",
        "backend-common",
        "common",
        "db-events",
        "ecosystem",
        "eslint-plugin-qatalog",
        "templates",
        "workflows"
      ]
    }
  ]
}
```

[log snippet](https://app.renovatebot.com/dashboard#github/qatalog/monorepo/702739773) that I believe shows `prConcurrentLimit = 10`, but no other PRs being opened:
```
DEBUG: Calculating hourly PRs remaining
DEBUG: getPrList success
{
  "pullsTotal": 20,
  "requestsTotal": 2,
  "apiQuotaAffected": true
}
DEBUG: currentHourStart=2022-06-16T04:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating prConcurrentLimit (10)
DEBUG: getBranchPr(renovate/mustache-4.x)
DEBUG: findPr(renovate/mustache-4.x, undefined, open)
DEBUG: Found PR #7979
DEBUG: getBranchPr(renovate/react-17.x)
DEBUG: findPr(renovate/react-17.x, undefined, open)
DEBUG: Found PR #7980
DEBUG: getBranchPr(renovate/react-dom-17.x)
DEBUG: findPr(renovate/react-dom-17.x, undefined, open)
DEBUG: Found PR #8058
DEBUG: getBranchPr(renovate/uuid-8.x)
DEBUG: findPr(renovate/uuid-8.x, undefined, open)
DEBUG: Found PR #8059
DEBUG: getBranchPr(renovate/async-3.x)
DEBUG: findPr(renovate/async-3.x, undefined, open)
DEBUG: findPr(renovate/async-3.x, undefined, closed)
DEBUG: getBranchPr(renovate/cross-fetch-3.x)
DEBUG: findPr(renovate/cross-fetch-3.x, undefined, open)
DEBUG: findPr(renovate/cross-fetch-3.x, undefined, closed)
```

**Comment by rarkins** [comment]
What makes you think it's related to the concurrent PR limit?

**Reply by leo-mendoza-qatalog** [reply]
Oh sorry, I think I expressed myself poorly - I think this is actually two issues clubbed into one (which makes me think my configuration is wrong overall).

1. renovatebot only targetting `@types/*`
2. prConcurrentLimit not getting respected

I'm fairly confident they aren't related as renovatebot was targetting just `@types` prior to our `prConcurrentLimit` change - this just happened to be another issue I noticed in my log parsing this morning.

**Reply by rarkins** [reply]
Is there anything you can point to in the debug log and say "I think this is where something went wrong"?

Have you looked at the "packageFiles with updates" log message for example?

**Reply by leo-mendoza-qatalog** [reply]
I do see some entries in there which I expect to get updated such as:

```
 {
            "depType": "resolutions",
            "depName": "async",
            "currentValue": "^3.2.3",
            "datasource": "npm",
            "prettyDepType": "resolutions",
            "lockedVersion": "3.2.3",
            "depIndex": 31,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "3.2.4",
                "newValue": "^3.2.4",
                "releaseTimestamp": "2022-06-07T03:59:17.813Z",
                "newMajor": 3,
                "newMinor": 2,
                "updateType": "patch",
                "isRange": true,
                "isBump": true,
                "branchName": "renovate/async-3.x"
              }
            ],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/caolan/async",
            "homepage": "https://caolan.github.io/async/",
            "currentVersion": "3.2.3",
            "isSingleVersion": false,
            "fixedVersion": "3.2.3"
          },
```

but I've also noticed a few other bits of weirdness like the labels or reviewers not being set so i'm happy to close this discussion for now and do a bit more debugging. I'll reopen once I have a more targeted question.

thanks for looking at this!

**Reply by rarkins** [reply]
I wanted you to look at that output because it tells you whether Renovate even "knows" there's updates to do. If Renovate is missing updates which you are expecting, then the problem is happening before you reach PR limits. For example it might be because you were expecting Renovate's default behavior to be rangeStrategy=update-lockfile (btw I still think this is the highest probability to solve your problem)

**Reply by leo-mendoza-qatalog** [reply]
oooh thanks for the hint, i will try this out!
